datalake/src/main/java/com/sequenceiq/datalake/flow/repair/SdxRepairActions.java

```diff
@@ -0,0 +1,154 @@
+package com.sequenceiq.datalake.flow.repair;
+
+import static com.sequenceiq.datalake.flow.repair.SdxRepairEvent.SDX_REPAIR_FAILED_HANDLED_EVENT;
+import static com.sequenceiq.datalake.flow.repair.SdxRepairEvent.SDX_REPAIR_FINALIZED_EVENT;
+import static com.sequenceiq.datalake.flow.repair.SdxRepairEvent.SDX_REPAIR_IN_PROGRESS_EVENT;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.datalake.entity.SdxClusterStatus;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.SdxEvent;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairFailedEvent;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairStartEvent;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairSuccessEvent;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairWaitRequest;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.SdxRepairService;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.flow.core.FlowEvent;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.flow.core.FlowState;
+import com.sequenceiq.notification.NotificationService;
+
+@Configuration
+public class SdxRepairActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxRepairActions.class);
+
+    @Inject
+    private SdxService sdxService;
+
+    @Inject
+    private SdxRepairService repairService;
+
+    @Inject
+    private NotificationService notificationService;
+
+    @Bean(name = "SDX_REPAIR_START_STATE")
+    public Action<?, ?> sdxDeletion() {
+        return new AbstractSdxAction<>(SdxRepairStartEvent.class) {
+
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext, SdxRepairStartEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, SdxRepairStartEvent payload, Map<Object, Object> variables) throws Exception {
```

Comment:
please add requestId to MDCContext in every doExecute call: `MDCBuilder.addRequestIdToMdcContext(context.getRequestId());`